### PR TITLE
Issue #1178: Update password strength meter

### PR DESCRIPTION
**Pull request by gterzian**
WIP for https://github.com/stellar/stellar-client/issues/1178

### CODE REVIEW

**Comment by gterzian** [comment]
@bekkibolthouse @bartekn @lowe 

How does this look as a first pass?

File: bower.json [hunk 1]
```diff
@@ -28,7 +28,8 @@
     "bignumber.js": "1.4.1",
     "angulartics": "0.17.1",
     "URIjs": "1.14.1",
-    "angular-recaptcha": "~2.1.1"
+    "angular-recaptcha": "~2.1.1",
+    "zxcvbn": "*"
```
Comment (bartekn) [inline bower.json]: Remove `angular-password-strength` from bower.json dependencies and from app dependencies.

Comment (bartekn) [inline bower.json]: Please specify semver version for `zxcvbn` package.

File: app/scripts/controllers/password-controller.js [hunk 2]
```diff
@@ -1,50 +1,15 @@
 'use strict';
+/* global zxcvbn*/
 
 var sc = angular.module('stellarClient');
 
-sc.controller('PasswordCtrl', function($scope, passwordStrengthComputations, badPasswords) {
```
Comment (bartekn) [inline app/scripts/controllers/password-controller.js]: Remove `bad_passwords.js` file.

**Comment by bartekn** [comment]
Some minor things needs to be changed. Looks great so far!

**Comment by gterzian** [comment]
@bartekn for some reason adding zxcvbn to bower only works with that wildcard, any ideas @lowe ?

**Comment by bartekn** [comment]
Bower is using Github tags when determining a version. I can see they have only [one version](https://github.com/dropbox/zxcvbn/releases) released a few years ago. You can try to pull a fixed version using commit id:
```
"zxcvbn": "git+https://github.com/dropbox/zxcvbn.git#{commit-id}"
```

**Comment by gterzian** [comment]
@bartekn All good on the zxcvbn semver, followed your suggestions. It seems the existing password files are used elsewhere in the app, the e2e tests started failing. I've re-added them for now, and I guess we can look further into other parts of the app where we can replace the old password logic with the new one using zxcvbn. Perhaps as part of another PR?

**Comment by gterzian** [comment]
@bartekn Ok actually it's some UI stuff in the template that is directly using the old password checks, will need to look further into that :smile:

**Comment by gterzian** [comment]
@bartekn @bekkibolthouse  Ok I had to replace a filter from `angular-password` for something using `zxcvbn`, looks good at this point. I did some limited manual QA locally, anyone care to join in the effort?

**Comment by gterzian** [comment]
@bartekn how does it look Bartek?

**Comment by bartekn** [comment]
Your PR looks fine right now and I think it's ready to be merged but I'd change one last thing. Of course everything below is not in a scope of this PR but we would be greatful if you could check it out.

So basically we calculate zxcvbn score 3 times: in `PasswordCtrl` (in `checkPassword()` and `passwordStrength()` functions) and in `passwordScore` filter.

A better way to do it would be to add a watch listener to `password` field which should:
* calculate `zxcvbn` score,
* add a word describing a score to the `$scope` so we don't have to execute `passwordStrength()` function,
* add a class name associated with a score to the `$scope` so we don't have to use `passwordScore` filter.

The biggest problem with a current code is that `passwordStrength` and filter are inlined in HTML:
```html
<div class="description">{{ passwordStrength() }}</div>
```
```html
<div class="password-strength {{ data.password | passwordScore }}" ng-class="{null: data.password == ''}">
```
and are executed even if password does not change. After implementing changes above we'll use `zxcvb` only when needed.

**Comment by gterzian** [comment]
@bartekn Ok thanks, that's a really good one, please take a look now

**Comment by bartekn** [comment]
Hey @gterzian, thanks for your work! I added some minor changes (eg. removed one unused function) but then I realized I can't push to your repo. So I'm going to merge your PR and then push my commit to master. Once again, thank you for fixing PasswordController.

**Pull request merged**